fe/fe-core/src/main/java/com/starrocks/persist/PartitionPersistInfoV2.java

```diff
@@ -0,0 +1,128 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.persist;
+
+import com.google.gson.annotations.SerializedName;
+import com.starrocks.catalog.DataProperty;
+import com.starrocks.catalog.Partition;
+import com.starrocks.common.io.Text;
+import com.starrocks.common.io.Writable;
+import com.starrocks.persist.gson.GsonUtils;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+public abstract class PartitionPersistInfoV2 implements Writable {
+
+    public static final  short UN_PARTITION_TYPE = 0;
+    public static final  short LIST_PARTITION_TYPE = 1;
+    public static final  short RANGE_PARTITION_TYPE = 2;
+
+    @SerializedName("dbId")
+    private Long dbId;
+    @SerializedName("tableId")
+    private Long tableId;
+    @SerializedName("partition")
+    private Partition partition;
+    @SerializedName("dataProperty")
+    private DataProperty dataProperty;
+    @SerializedName("replicationNum")
+    private short replicationNum;
+    @SerializedName("isInMemory")
+    private boolean isInMemory;
+    @SerializedName("isTempPartition")
+    private boolean isTempPartition;
+
+    public PartitionPersistInfoV2() {
+    }
+
+    public PartitionPersistInfoV2(Long dbId, Long tableId, Partition partition,
+                                  DataProperty dataProperty, short replicationNum,
+                                  boolean isInMemory, boolean isTempPartition) {
+        this.dbId = dbId;
+        this.tableId = tableId;
+        this.partition = partition;
+        this.dataProperty = dataProperty;
+        this.replicationNum = replicationNum;
+        this.isInMemory = isInMemory;
+        this.isTempPartition = isTempPartition;
+    }
+
+    public final boolean isListPartitionPersistInfo() {
+        return this.getClass() == ListPartitionPersistInfo.class;
+    }
+
+    public final boolean isRangePartitionPersistInfo() {
+        return this.getClass() == RangePartitionPersistInfo.class;
+    }
+
+    public final ListPartitionPersistInfo asListPartitionPersistInfo() {
+        return (ListPartitionPersistInfo) this;
+    }
+
+    public final RangePartitionPersistInfo asRangePartitionPersistInfo() {
+        return (RangePartitionPersistInfo) this;
+    }
+
+    public abstract PartitionPersistInfoV2 readFieldIn(DataInput in) throws IOException;
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        if (this.isListPartitionPersistInfo()) {
+            out.writeShort(LIST_PARTITION_TYPE);
+        } else if (this.isRangePartitionPersistInfo()) {
+            out.writeShort(RANGE_PARTITION_TYPE);
+        } else {
+            out.writeShort(UN_PARTITION_TYPE);
+        }
+        String json = GsonUtils.GSON.toJson(this);
+        Text.writeString(out, json);
```

Comment:
Better to use Gson serialization directly.
You can refer to `Resource, SparkResource, HiveResource`.